### PR TITLE
Skip VH polymorphic method when resolving MH ref

### PR DESCRIPTION
**Pull request by fengxue-IS**
VH polymorphic methods have special handling in
MethodHandleNatives.linkMethodHandleConstant()
and doesn't require CP entry to be resolved

Related: #12539, #13094

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
@tajila can you please take a look
This PR fixes `java/lang/invoke/VarHandles/VarHandleMethodReferenceTest.java`

File: runtime/vm/resolvesupport.cpp [hunk 1]
```diff
@@ -1555,93 +1558,8 @@ resolveVirtualMethodRefInto(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA c
 			J9UTF8 *sigUTF = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
 			U_8* initialMethodName = J9UTF8_DATA(nameUTF);
 			U_16 initialMethodNameLength = J9UTF8_LENGTH(nameUTF);
-			BOOLEAN isVarHandle = FALSE;
 
-			switch (initialMethodNameLength) {
```
Comment (tajila) [inline runtime/vm/resolvesupport.cpp:1560]: do we still need this for the non-J9VM_OPT_OPENJDK_METHODHANDLE code?

Comment (fengxue-IS) [reply runtime/vm/resolvesupport.cpp:1560]: VM_VMHelpers::isPolymorphicVarHandleMethod has the same functionality, I opt to use the helper to reduce duplicate code in the VM

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity win jdk16,jdk11

**Comment by tajila** [comment]
jenkins test sanity,extended xlinuxojdk292 jdk16

**Comment by tajila** [comment]
jenkins test sanity xlinuxojdk292 jdk16

**Pull request merged**